### PR TITLE
Fix issue 4210

### PR DESCRIPTION
**Pull request by botaberg**
Fix build warnings in ported interop tests by removing includes of platformdefines.h.

### CODE REVIEW

**Comment by botaberg** [comment]
@tijoytom @yizhang82 PTAL

File: tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.h [hunk 1]
```diff
@@ -358,7 +357,13 @@ bool IsCorrectCharSetUnicodeSequential(CharSetUnicodeSequential* p)
 	LPCWSTR expected= u"some string";
 #endif
 	LPCWSTR actual = p->f1;
-	if(0 != TP_wcmp_s((WCHAR*)actual, (WCHAR*)expected))
+	int lenExpected = wcslen(expected);
+	int lenActual = wcslen(actual);
```
Comment (yizhang82) [inline tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.h]: This is where platformdefine.cpp would've helped.

**Comment by yizhang82** [comment]
@tijoytom @botaberg We should find out why we have duplicated x-plat functions/defines in xplatform.h vs platformdefines.h. there is a good reason we are diverging?

**Comment by tijoytom** [comment]
@yizhang82 Interop tests have no dependency outside interop folder now  and we want it to be that way since eventually these tests might move to different repository. In fact platformdefine it self is mostly copy from PAL.

**Comment by yizhang82** [comment]
@tijoytom If they are copy from PAL, we should be fine. Let's keep an eye on future issues (as matter of fact, I have fixed a bug in xplatform.h) and see if we need to change course. My worry is that we'll end up maintaining separate xplat library and will not benefit from the major bug fixes. It also doesn't help that our native tests would look very different from rest of the native tests. But in practice, this probably doesn't matter much, but I'd hoped we've taken a different approach. If we worry about taking it to a different repo, we could simply take a copy of platformdefines.h/.cpp to that repo and it wouldn't be an issue (unless platformdefines.h/.cpp takes in a ton of dependencies, but that doesn't seem to be the case)

**Comment by botaberg** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please

**Comment by botaberg** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test please

**Comment by danmosemsft** [comment]
@dotnet-bot test this please

**Comment by danmosemsft** [comment]
@botaberg this seems to have introduced an error:

```17:24:05 In file included from /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx_prtest/tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.cpp:1:
17:24:05 
/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx_prtest/tests/src/Interop/StructMarshalling/PInvoke/MarshalStructAsParamDLL.h:342:3: error: use of undeclared identifier 'wcscpy_s'; did you mean 'wcsncpy_s'?
17:24:05                 wcscpy_s((WCHAR*)temp, (len+1)*sizeof(WCHAR), strSource);
17:24:05                 ^~~~~~~~
17:24:05                 wcsncpy_s
17:24:05 /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx_prtest/tests/src/Interop/common/xplatform.h:186:5: note: 'wcsncpy_s' declared here
17:24:05 int wcsncpy_s(LPWSTR strDestination, size_t size1, LPCWSTR strSource)
```

Are you able to take a look?

**Comment by danmosemsft** [comment]
@botaberg I'm going to close this since the issue is apparently fixed -- https://github.com/dotnet/coreclr/issues/4210

**Pull request closed without merging**